stablehlo/reference/Ops.cpp

```diff
@@ -199,10 +200,10 @@ SmallVector<Tensor> split(const Tensor &x, int64_t numResults, Axis axis,
 
 }  // namespace
 
-SmallVector<InterpreterValue> eval(
-    Region &region, ArrayRef<InterpreterValue> args, Process *process,
-    Scope *parent,
-    llvm::function_ref<llvm::Error(Operation &, Process *, Scope &)> fallback) {
+SmallVector<InterpreterValue> eval(Region &region,
+                                   ArrayRef<InterpreterValue> args,
+                                   const InterpreterConfiguration *config,
```

Comment:
Why not make this a reference? It looks like the important part is that Fallback is optional. I think making config optional and fallback optional makes the code more complicated.

```cpp
if (config && !config->fallback)
==>
if (!config->fallback)

foo(...,/*config=*/nullptr)
==>
foo(..., /*config=*/{})
```

Reply:
Also for the places where config is nullptr.. is this the proper behavior? I feel like this should be passed through regions? I.e. probe in the body of a `stablehlo.map` operation looks like it wouldn't work currently

Reply:
Yup, https://github.com/openxla/stablehlo/pull/1797#discussion_r1378996442 is the reason this is an optional pointer for the time being, since otherwise the other interfaces that call into `eval` today would need to be updated to take a `config` as described in #1820. I think maybe this could be changed to required pass by reference once those ops are updated to take the fallback interface correctly.